src/fdb5/api/FDB.h

```diff
@@ -72,73 +76,200 @@ class FDB {
     FDB(FDB&&)            = default;
     FDB& operator=(FDB&&) = default;
 
-    // -------------- Primary API functions ----------------------------
+    // -------------- Primary API functions ----------------------------------------------------------------------------
 
+    /// Archive binary data to a FDB.
+    /// Any callback set with registerArchiveCallback will be invoked.
+    /// @param handle eckit::message::Message to data to archive
```

Comment:
Not just binary data. An eckit::Message.

This is data which has been pre-chunked into the objects to be stored. And that we know is self-describing (hence not needing a key to be supplied)